src/liberty/devCommands.ts

```diff
@@ -610,4 +612,44 @@ async function getLocalGradleWrapper(projectFolder: string): Promise<string | un
  */
 function isWin(): boolean {
     return process.platform.startsWith("win");
-}
\ No newline at end of file
+}
+
+/*
+will return the path of the report, since there are diffrent folders to look into and the file names can be different 
+we need to get the paths to look for dynamically
+*/
+function getReportFile(path : any, dir : string, filename : string) : any{
+    return Path.join(path, "target", dir, filename);
+}
+
+/*
+Function will check if the report is available within the given path and returns a boolean based on it and also 
+  the report will be displayed if it is available
+*/
+function checkReportAndDisplay(report : any, reportType : string, reportTypeLabel : string, libProject : LibertyProject, directory : string) : Promise<boolean> {
+    return new Promise((resolve) => {
+      fs.exists(report, (exists) => {
+        if(exists){
+            const panel = vscode.window.createWebviewPanel(
+                reportType, // Identifies the type of the webview. Used internally
+                libProject.getLabel() + " " + reportTypeLabel + " report", // Title of the panel displayed to the user
+                vscode.ViewColumn.Two, // Open the panel in the second window
+                {}, // Webview options
+            );
+            panel.webview.html = getReport(report); // display HTML content
+            /*
+            For maven projects we need to check for the report in 'reports' and 'site', we only need to show the message if it is not 
+            available in both the locations, below condition make sure to avoid the message when its a maven project and the directory 
+            is 'reports'
+            */ 
+        }else if(!((libProject.getContextValue() === LIBERTY_MAVEN_PROJECT || libProject.getContextValue() === LIBERTY_MAVEN_PROJECT_CONTAINER) 
+                    && directory=="reports")){
+
+                const message = localize("test.report.does.not.exist.run.test.first", report);
+                vscode.window.showInformationMessage(message);
+        }
+        console.log("report available::"+exists);
```

Comment:
Do we need this log statement? Same as above,  if you think it is useful, it is probably best as a debug statement.